spring-integration-mqtt/src/main/java/org/springframework/integration/mqtt/inbound/Mqttv5PahoMessageDrivenChannelAdapter.java

```diff
@@ -215,8 +223,12 @@ public void destroy() {
 	public void addTopic(String topic, int qos) {
 		this.topicLock.lock();
 		try {
-			this.mqttClient.subscribe(topic, qos).waitForCompletion(getCompletionTimeout());
 			super.addTopic(topic, qos);
+			if (mqttClient != null && mqttClient.isConnected()) {
+				this.mqttClient.subscribe(topic, qos).waitForCompletion(getCompletionTimeout());
+			} else {
+				logger.warn(String.format("An attempt to add topic : '%s' with QoS : '%s' without having mqtt client initialized or connected", topic, qos));
```

Comment:
This one has to be removed, too.
It is fully OK to add topics which could be subscribed later on, when `start()` is called.